### PR TITLE
rbd-mirror: integrate image map policy as incremental step to active-active

### PR DESCRIPTION
**Pull request by dillaman**

### CODE REVIEW

**Comment by dillaman** [comment]
http://pulpito.ceph.com/jdillaman-2018-04-09_07:49:53-rbd-wip-rbd-mirror-active-active-testing-basic-smithi/

File: src/tools/rbd_mirror/Instances.cc [hunk 1]
```diff
@@ -89,14 +90,57 @@ void Instances<I>::handle_acked(const InstanceIds& instance_ids) {
...
+  Mutex::Locker locker(m_lock);
+  InstanceIds added_instance_ids;
+  for (auto& instance_id : instance_ids) {
+    auto it = m_instances.find(instance_id);
+    if (it != m_instances.end() && it->second.state == INSTANCE_STATE_ADDING) {
+      added_instance_ids.push_back(instance_id);
+    }
+  }
+
+  dout(5) << "instance_ids=" << added_instance_ids << dendl;
```
Comment (trociny) [inline src/tools/rbd_mirror/Instances.cc]: Probably skip the part below if `added_instance_ids` happens to be empty?

File: src/tools/rbd_mirror/Instances.cc [hunk 2]
```diff
@@ -89,14 +90,57 @@ void Instances<I>::handle_acked(const InstanceIds& instance_ids) {
...
+  m_lock.Unlock();
+  m_listener.handle_added(added_instance_ids);
+  m_lock.Lock();
+
+  for (auto& instance_id : instance_ids) {
```
Comment (trociny) [inline src/tools/rbd_mirror/Instances.cc]: Shouldn't we iterate over `added_instance_ids` here?

**Comment by trociny** [comment]
@dillaman it needs rebase after #21080 merged

**Comment by trociny** [comment]
@ceph-jenkins retest this please

**Pull request merged**